modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2703,6 +2703,28 @@ def explode(self, axis: Union[int, Axis], func: Callable) -> "PandasDataframe":
             partitions, new_index, new_columns, row_lengths, column_widths
         )
 
+    def to_pandas_in_remote_function(self) -> "PandasDataframe":
+        """
+        Create a PandasFrame with single partition from the partitions of the current dataframe.
+
+        Returns
+        -------
+        PandasDataframe
+            An PandasDataframe containing only a single partition.
+        """
+        partition_with_to_pandas_data = [
+            self._partition_mgr_cls.convert_partitions_to_pandas_in_remote_func(
+                self._partitions
+            )
+        ]
+        partitions = np.array(partition_with_to_pandas_data).reshape(1, -1)
+
+        result = self.__constructor__(
+            partitions, self.index, columns=self.columns, dtypes=self.dtypes
+        )
+        result.synchronize_labels()
```

Comment:
Do we really need this?

Reply:
Yes, the index could be used as "on" parameter in merge if, the indexes are not synchronized when the result dataframe broadcasted it would not have the index information in the remote function causing an error.

Reply:
I see, thanks!